Strengthen property tests for `applyBlocks`.

jonathanknowles opened the pull request:
## Issue

ADP-2840 / #3785

## Summary

The `applyBlocks` function was recently changed so that fees of incoming transactions, if present (i.e., not `Nothing`), are no longer stripped away (i.e., no longer replaced with `Nothing`).

The following commits are relevant:

- https://github.com/input-output-hk/cardano-wallet/commit/120c7decdc662607758de60c7c8690845baeb22b
- https://github.com/input-output-hk/cardano-wallet/commit/634265eb4bb13bc0c7782c889753affce62e1760

Consequently, it is now more straightforward to test our expectations about the `fee` field within our property tests for `applyBlocks`. In particular, we now test that `applyBlocks` does not modify the `fee` field in any way (if it is not `Nothing`), whereas before, the `fee` field was deliberately excluded from the test.

Note that this does rely on the fact that our transaction sequence generator never generates transactions with fees that are `Nothing`:

https://github.com/input-output-hk/cardano-wallet/blob/4af4c184a2f986a51c44c0f93a5eef7e7fa31acf/lib/wallet/src/Cardano/Wallet/Primitive/Types/Tx/TxSeq/Gen.hs#L319-L320

If in future we revise this generator to also cover transactions where fees can be `Nothing`, we will have to also revise these property tests.

This PR adjusts the following property tests, which test that `applyBlocks` returns all transactions that are relevant to the wallet (and no others), and that they are returned in the same order that they appear in the blockchain:

- [`prop_applyBlocks_filteredTxs_someOurs`](https://github.com/input-output-hk/cardano-wallet/blob/76f09ef7fda58418710b8394603d75d677b34425/lib/wallet/test/unit/Cardano/Wallet/Primitive/ModelSpec.hs#L2657)
    tests `applyBlocks` in scenarios where **_only some_** transactions in an arbitrary block sequence are relevant to a wallet.
- [`prop_applyBlocks_filteredTxs_allOurs`](https://github.com/input-output-hk/cardano-wallet/blob/76f09ef7fda58418710b8394603d75d677b34425/lib/wallet/test/unit/Cardano/Wallet/Primitive/ModelSpec.hs#L2687)
    tests `applyBlocks` in scenarios where **_all_** transactions in an arbitrary block sequence are relevant to a wallet.

jonathanknowles committed:
Strengthen `prop_applyBlocks_filteredTxs_someOurs`.

The `applyBlocks` function was recently changed so that fees of incoming
transactions, if present (i.e., not `Nothing`), are no longer stripped
away (i.e., no longer replaced with `Nothing`).

The following commits are relevant:

120c7decdc662607758de60c7c8690845baeb22b
634265eb4bb13bc0c7782c889753affce62e1760

Consequently, it is now more straightforward to test our expectations
about the `fee` field within our property tests for `applyBlocks`. In
particular, we now test that `applyBlocks` does not modify the `fee`
field in any way (if it is not `Nothing`), whereas before, the `fee`
field was deliberately excluded from the test.

Note that this does rely on the fact that our transaction sequence
generator never generates transactions with fees that are `Nothing`:

https://github.com/input-output-hk/cardano-wallet/blob/4af4c184a2f986a51c44c0f93a5eef7e7fa31acf/lib/wallet/src/Cardano/Wallet/Primitive/Types/Tx/TxSeq/Gen.hs#L320

If in future we revise this generator to also cover transactions where
fees can be `Nothing`, we will have to also revise these property tests.

This commit strengthens the `prop_applyBlocks_filteredTxs_someOurs`
property, which tests the `applyBlocks` function in scenarios where
**_only some_** transactions in an arbitrary block sequence are
relevant to an arbitrary wallet.

jonathanknowles committed:
Strengthen `prop_applyBlocks_filteredTxs_allOurs`.

The `applyBlocks` function was recently changed so that fees of incoming
transactions, if present (i.e., not `Nothing`), are no longer stripped
away (i.e., no longer replaced with `Nothing`).

The following commits are relevant:

120c7decdc662607758de60c7c8690845baeb22b
634265eb4bb13bc0c7782c889753affce62e1760

Consequently, it is now more straightforward to test our expectations
about the `fee` field within our property tests for `applyBlocks`. In
particular, we now test that `applyBlocks` does not modify the `fee`
field in any way (if it is not `Nothing`), whereas before, the `fee`
field was deliberately excluded from the test.

Note that this does rely on the fact that our transaction sequence
generator never generates transactions with fees that are `Nothing`:

https://github.com/input-output-hk/cardano-wallet/blob/4af4c184a2f986a51c44c0f93a5eef7e7fa31acf/lib/wallet/src/Cardano/Wallet/Primitive/Types/Tx/TxSeq/Gen.hs#L320

If in future we revise this generator to also cover transactions where
fees can be `Nothing`, we will have to also revise these property tests.

This commit strengthens the `prop_applyBlocks_filteredTxs_allOurs`
property, which tests the `applyBlocks` function in scenarios where
**_all_** transactions in an arbitrary block sequence are relevant to
an arbitrary wallet.

piotr-iohk reviewed (approved):
So it seems that https://github.com/input-output-hk/cardano-wallet/pull/3806 is changing the behavior only for txs that are made in Byron era. `fees` are still shown on incoming and outgoing txs for all txs post-Byron, therefore I believe it makes sense to have this change.

jonathanknowles committed:
Add `TODO` to transaction sequence generator. (ADP-2895)

This TODO highlights the fact that we currently only ever generate
transactions where fees are not `Nothing`.

See: https://input-output.atlassian.net/browse/ADP-2895
